core/common/src/main/java/alluxio/network/protocol/RPCFileReadResponse.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol;
+
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.network.protocol.databuffer.DataByteBuffer;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Longs;
+import com.google.common.primitives.Shorts;
+import io.netty.buffer.ByteBuf;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.nio.ByteBuffer;
+
+/**
+ * This represents the response of a {@link RPCFileReadRequest}.
+ */
+@ThreadSafe
+public final class RPCFileReadResponse extends RPCResponse {
+  private final long mTempUfsFileId;
+  private final long mOffset;
+  private final long mLength;
+  private final DataBuffer mData;
```

Comment:
I guess we are relying on TCP checksum to do error detection.  Have you guys seen any data corruption in the past?